src/plugins/console/public/application/containers/editor/legacy/console_editor/mapbox_vector_tile/convert_mapbox_vector_tile_to_json.test.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { readFileSync } from 'fs';
+import { resolve } from 'path';
+import { VectorTile } from '@mapbox/vector-tile';
+import Protobuf from 'pbf';
+import { convertMapboxVectorTileToJson } from './convert_mapbox_vector_tile_to_json';
+
+const PATH_TO_PBF = resolve(__dirname, './response.pbf'); // Query sample "GET kibana_sample_data_logs/_mvt/geo.coordinates/8/47/98"
+
+describe('Convert mapbox vector tile to json', () => {
+  let vectorTile: VectorTile;
+
+  beforeAll(() => {
+    const response = readFileSync(PATH_TO_PBF);
+    vectorTile = new VectorTile(new Protobuf(response));
+  });
+
+  it('function should convert vectorTile to string', () => {
+    expect(() => {
+      const json = convertMapboxVectorTileToJson(vectorTile);
+      JSON.parse(json);
+    }).not.toThrow();
+  });
+
+  it('response should have [meta, aggs, hits] layers', () => {
+    const vectorTileJson = JSON.parse(convertMapboxVectorTileToJson(vectorTile));
+
+    expect(vectorTileJson).toHaveProperty('meta');
```

Comment:
I'm not really sure it's a good idea to compare to a full object, just because the typical response of vector tile data is a pretty big json object. 
Here's an example of it, and this is just for one specific coordinate like `8/47/98`:
<details>
<summary>Click to expand</summary>

```json
{
  "meta":[
    {
      "geometry":{
        "type":"Polygon",
        "coordinates":[
          [
            [
              0,
              4096
            ],
            [
              4096,
              4096
            ],
            [
              4096,
              0
            ],
            [
              0,
              0
            ],
            [
              0,
              4096
            ]
          ]
        ]
      },
      "properties":{
        "_shards.failed":0,
        "_shards.skipped":0,
        "_shards.successful":1,
        "_shards.total":1,
        "aggregations._count.avg":5,
        "aggregations._count.count":3,
        "aggregations._count.max":8,
        "aggregations._count.min":3,
        "aggregations._count.sum":15,
        "hits.total.relation":"eq",
        "hits.total.value":15,
        "timed_out":false,
        "took":67
      }
    }
  ],
  "hits":[
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          3585,
          2204
        ]
      },
      "properties":{
        "_id":"-BbN9X8BEs3pqeJ47-oU",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          3175,
          3576
        ]
      },
      "properties":{
        "_id":"QhbN9X8BEs3pqeJ47-sU",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          3585,
          2204
        ]
      },
      "properties":{
        "_id":"GhfN9X8BEs3pqeJ49QJE",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          3175,
          3576
        ]
      },
      "properties":{
        "_id":"uhfN9X8BEs3pqeJ49w44",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          3585,
          2204
        ]
      },
      "properties":{
        "_id":"bhfN9X8BEs3pqeJ49xP6",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          3585,
          2204
        ]
      },
      "properties":{
        "_id":"XhfN9X8BEs3pqeJ4-iNJ",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          2603,
          1476
        ]
      },
      "properties":{
        "_id":"LxfN9X8BEs3pqeJ4-ykP",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          2603,
          1476
        ]
      },
      "properties":{
        "_id":"mxfN9X8BEs3pqeJ4-zDa",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          3585,
          2204
        ]
      },
      "properties":{
        "_id":"dhfN9X8BEs3pqeJ4_zlP",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          3585,
          2204
        ]
      },
      "properties":{
        "_id":"DRfN9X8BEs3pqeJ4_zpQ",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          3175,
          3576
        ]
      },
      "properties":{
        "_id":"ZxfO9X8BEs3pqeJ4Akf1",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          3585,
          2204
        ]
      },
      "properties":{
        "_id":"-xfO9X8BEs3pqeJ4Akf1",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          3585,
          2204
        ]
      },
      "properties":{
        "_id":"fxfO9X8BEs3pqeJ4ClfI",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          2603,
          1476
        ]
      },
      "properties":{
        "_id":"fxfO9X8BEs3pqeJ4C1rB",
        "_index":"kibana_sample_data_logs"
      }
    },
    {
      "geometry":{
        "type":"Point",
        "coordinates":[
          2603,
          1476
        ]
      },
      "properties":{
        "_id":"thfO9X8BEs3pqeJ4DFvc",
        "_index":"kibana_sample_data_logs"
      }
    }
  ],
  "aggs":[
    {
      "geometry":{
        "type":"Polygon",
        "coordinates":[
          [
            [
              3584,
              2208
            ],
            [
              3600,
              2208
            ],
            [
              3600,
              2192
            ],
            [
              3584,
              2192
            ],
            [
              3584,
              2208
            ]
          ]
        ]
      },
      "properties":{
        "_key":"16/12256/25225",
        "_count":8
      }
    },
    {
      "geometry":{
        "type":"Polygon",
        "coordinates":[
          [
            [
              2592,
              1488
            ],
            [
              2608,
              1488
            ],
            [
              2608,
              1472
            ],
            [
              2592,
              1472
            ],
            [
              2592,
              1488
            ]
          ]
        ]
      },
      "properties":{
        "_key":"16/12194/25180",
        "_count":4
      }
    },
    {
      "geometry":{
        "type":"Polygon",
        "coordinates":[
          [
            [
              3168,
              3584
            ],
            [
              3184,
              3584
            ],
            [
              3184,
              3568
            ],
            [
              3168,
              3568
            ],
            [
              3168,
              3584
            ]
          ]
        ]
      },
      "properties":{
        "_key":"16/12230/25311",
        "_count":3
      }
    }
  ]
}
```
</details>

Reply:
> I think this will also address 80% of your request to add a functional test. Functional tests are slower than unit tests and the Console ones have been especially flaky, so maybe we don't need to add functional test coverage if we enhance these units tests.

I am not as concerned with the output format as ensuring vector tile responses continue to work. There are lots of things that could break this that are not tied to the formatting of a vector tile to an output such as blocking/removing http header. The use case for running `_mvt` API commands in console is narrow so I think having a functional test to ensure everything continues to work adds a lot of value and is worth test maintenance costs. 

> I'm not really sure it's a good idea to compare to a full object, just because the typical response of vector tile data is a pretty big json object.

How about adding a query to the sample request so the tile only contains a single hit?

Reply:
Those are good points, Nathan. I'm +1 on giving a functional test a shot, but I suggest we run it through the flaky test runner. If it turns out to be flaky we can fall back to the unit test.